### ISSUE TITLE
remove the default datetime type conversion

### ISSUE THREAD

**Issue by freefallzhou** [FeatureLabs/featuretools#191]
### Feature Request Description

When create an entity from dataframe, by default, featuretools will try to convert every non-type-specified column to date_time.

This operation based on pandas' to_datetime  function. see featuretools/entityset/entity.py line 331.

Cause this function will always throw exceptions when trying to convert text that may looks like date string in some situation(such as Numeric text, x-x-x-x format labels) but others are not.

I'm suggesting remove the default tries to covert to datetime. And  emphasize in the doc that datetype should be explicitly defined.

-----
*Issues created here on Github are for bugs or feature requests. For usage questions and questions about errors, please ask on Stack Overflow with the [featuretools](https://stackoverflow.com/questions/tagged/featuretools) tag. Check the [documentation](https://docs.featuretools.com/#get-help) for further guidance on where to ask your question.*

**Comment by kmax12** [comment]
@freefallzhou this make sense and a change we are considering. 

For the time being, you can specify the variable type of a column when calling `entity_from_dataframe` to avoid Featuretools trying to infer the variable type.

**Comment by Seth-Rothschild** [comment]
This can also show up in the `mock_customer_demo` if the first 10 values of `zip_code` are `"02139"`. Infer_variable_types will parse those as `01-01-2139` and then `entity_from_dateframe` will error when it tries to parse the other zip code (`60091`).

**Comment by 786country** [comment]
Just following up when this will be resolved?

**Comment by ctduffy** [comment]
Just following up. I think we will close this issue, as this seems to no longer be a problem with the most recent version of featuretools (currently 0.14.0). I was able to reproduce this error with the version 0.2.1, which was the latest release at the time of this issue, but not able to reproduce it with the most recent one. Furthermore, since that version from 2018, there have been many updates to the variable inference conversion that makes it much more robust and, in particular, unlikely to guess zipcodes as datetime objects. I have attatched two screenshots below. The first is the zipcodes being guessed as datetime objects with the 0.2.1 release:
<img width="927" alt="0.2.1 release" src="https://user-images.githubusercontent.com/14035415/82566655-3bb7ae00-9b4a-11ea-8b21-91151c6a5812.png">
And the second is the current performance:
<img width="925" alt="0.14.0 release" src="https://user-images.githubusercontent.com/14035415/82566825-7883a500-9b4a-11ea-872d-a8a651f4029c.png">
